crates/accelerate/src/results/marginalization.rs

```diff
@@ -152,19 +152,12 @@ pub fn marginal_memory(
             .collect()
     };
     if return_int {
-        if out_mem.len() < parallel_threshold || !run_in_parallel {
-            Ok(out_mem
-                .iter()
-                .map(|x| BigUint::parse_bytes(x.as_bytes(), 2).unwrap())
-                .collect::<Vec<BigUint>>()
-                .to_object(py))
-        } else {
-            Ok(out_mem
-                .par_iter()
-                .map(|x| BigUint::parse_bytes(x.as_bytes(), 2).unwrap())
-                .collect::<Vec<BigUint>>()
-                .to_object(py))
-        }
+        let int_pyobject = py.import("builtins")?.getattr("int")?;
+        Ok(out_mem
+            .iter()
+            .map(|x| int_pyobject.call1((x, 2u8)).unwrap())
+            .collect::<Vec<_>>()
+            .to_object(py))
```

Comment:
Depending on when we merge this, there's a PR on PyO3 to add support for `num-bigint`+`abi3` in the works: PyO3/pyo3#3198.